modules/services/utils/src/main/java/au/com/shiftyjelly/pocketcasts/utils/extensions/Intent.kt

```diff
@@ -0,0 +1,14 @@
+package au.com.shiftyjelly.pocketcasts.utils.extensions
+
+import android.content.Intent
+import android.os.Build
+import android.os.Parcelable
+
+inline fun <reified T : Parcelable> Intent.getParcelableExtraCompat(key: String): T? {
+    return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+        getParcelableExtra(key, T::class.java)
+    } else {
+        @Suppress("DEPRECATION")
+        getParcelableExtra(key) as? T
+    }
+}
```

Comment:
There already exists `androidx.core.content.IntentCompat.getParcelableExtra()` function that does exactly that.

Reply:
This method is deprecated
<img width="661" alt="image" src="https://github.com/user-attachments/assets/4eabc332-872c-4cde-ad2c-d272ee06ffde">

Reply:
You're looking at the wrong method. You're looking at `android.content.Intent.getParcelableExtra()`. I linked the one from the compat library - `androidx.core.content.IntentCompat.getParcelableExtra()`.

Reply:
Got it! Changed here [3dd70a5](https://github.com/Automattic/pocket-casts-android/pull/2780/commits/3dd70a5ceec4829f2c54a620ee025761f8d5bac4)